analyzers/tests/SonarAnalyzer.Test/TestCases/ApiControllersShouldDeriveFromController.cs

```diff
@@ -0,0 +1,265 @@
+
+        public object Foo4() => ViewData["foo"];
+        public object Foo5() => ViewBag["foo"];
+        public object Foo6() => TempData["foo"];
+    }
+
+    // TODO: Hey implementer, please extract these "InXXX" one-liners to a [DataTestMethod] test.
+    [ApiController]
+    public class InConstructor : Controller              // Compliant
+    {
+        object foo;
+        public InConstructor() => foo = View();
+    }
+
+    [ApiController]
+    public class InDestructor : Controller               // Compliant
+    {
+        object foo;
+        ~InDestructor() => foo = View();
+    }
+
+    [ApiController]
+    public class InPropertyGet : Controller              // Compliant
+    {
+        object foo => View();
+    }
+
+    [ApiController]
+    public class InPropertySet : Controller              // Compliant
+    {
+        object foo { set => _ = View(); }
+    }
+
+    [ApiController]
+    public class InIndexer : Controller                 // Compliant
+    {
+        object this[int index] => View();
+    }
+
+    [ApiController]
+    public class OverrideViewInvocation : Controller    // Compliant
+    {
+        public object Foo() => this.View();             // overrides and uses View
+        public override ViewResult View() => null;
+    }
+
+    [ApiController]
+    public class FakeViewInvocation : Controller        // FN, it's not an actual view
+    {
+        public object Foo() => this.View();             // hides View, does not override it
+        public ViewResult View() => null;
+    }
+
+    [ApiController]
+    public class BaseViewInvocation : Controller        // Compliant
+    {
+        public object Foo() => base.View();             // uses Controller.View
+        public ViewResult View() => null;
+    }
+
+    [ApiController]
+    public partial class Partial { }
+    public partial class Partial : Controller { }       // Compliant
+    public partial class Partial
+    {
+        public object Foo() => View();
+    }
+
+    [ApiController]
+    public class MemberReference : Controller               // Compliant
+    {
+        public Func<ViewResult> NotCalled() => this.View;   // nothing is invoked, but the dependency is used.
+    }
+
+    [ApiController]
+    public class NameOf : Controller          // Compliant
+    {
+        public object Foo() => nameof(View); // same as above
+    }
+
+    [ApiController]
+    public class PassByName : Controller                // Compliant
+    {
+        public void Foo() => ExpectsAction(View);
+        public void ExpectsAction(Func<ViewResult> func)
+        {
+            // here func could be invoked or not.
+        }
+    }
+
+    [ApiController]
+    public class PassByLambda : Controller              // Compliant
+    {
+        public void Foo() => ExpectsAction(() => View());
+        public void ExpectsAction(Func<ViewResult> func)
+        {
+            // here func could be invoked or not.
+        }
+    }
+}
```

Comment:
I did not add the tests because base on this line https://github.com/SonarSource/sonar-dotnet/blob/a4d269158f51d9840f9dbbe9c9730944cb9ac391/analyzers/src/SonarAnalyzer.CSharp/Rules/ApiControllersShouldDeriveFromController.cs#L62
we will never reach the baseList part and I cannot inherit from two classes to force it :/